fish_completions.go

```diff
@@ -207,11 +234,13 @@ complete -c %[2]s -e
 
 # The call to __%[1]s_prepare_completions will setup __%[1]s_comp_results
 # which provides the program's completion choices.
-complete -c %[2]s -n '__%[1]s_prepare_completions' -f -a '$__%[1]s_comp_results'
-
+# if this doesn't require order preservation, we dont use the -k flag
+complete -c %[2]s -n '__%[1]s_doesnt_requires_order_preservation' -n '__%[1]s_prepare_completions' -f -a '$__%[1]s_comp_results'
```

Comment:
This wasn't working for me.  After troubleshooting I found out that my version of fish didn't support using multiple conditions (`-n` flag).  Apparently this was added/fixed in fish 3.5: https://github.com/fish-shell/fish-shell/pull/8967

So, this change wouldn't work for older fish version.  But good news, we can replace the multiple `-n` logic with a single `-n` using a logical `&&`.  So we can instead do:
```
complete -c %[2]s -n '__%[1]s_doesnt_requires_order_preservation && __%[1]s_prepare_completions' -f -a '$__%[1]s_comp_results'
```